### ISSUE TITLE
Allow pipelining into `return`

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#13738]
### Related problem

Nushell uses a lot of functional elements, so I use pipelines a lot.  One issue which often comes up, however, is that you can't return from a pipeline.  So, you have to do awkward things with quoting, a la:

```Nushell
                return ($url
                    | update path {
                        split row "/"
                        | skip 1
                        | if ($in | length) > 2 {
                            update 2 raw
                        } else {
                            append [raw branch $branch]
                        }
                        | append Cargo.toml
                        | str join "/"
                        | "/" + $in
                    })
``` 

### Describe the solution you'd like

I think it would be convenient to be able to pipe into `return` to terminate a function:

```Nushell
$url.path
| split row "/"
| return  # returns the piped in list
```

### Describe alternatives you've considered

I sometimes write it as `... | return $in`, but it looks hacky, and I think it discourages using such a functional style return.

**Comment by fdncred** [comment]
I don't think this would be difficult. You just have to check to see if you have piped in input or parameterized input and do the right thing for each. Not sure what you do when there's both, maybe throw an error?

**Comment by KAAtheWiseGit** [comment]
Yeah, I was thinking an error if both are passed.  Not sure about the type, though (since there's an effort about not adding too many new errors)

**Comment by NotTheDr01ds** [comment]
Since `return` is also a parser keyword, would this be a similar implementation to piping to `let` as mentioned in #11951?

It reminds me of the `var` discussion since both would solve similar ergonomic problems.

Side-note:

> I think it discourages using such a functional style return.

Not sure I understand your concern there.  I absolutely love ...

```
| <command> $in
```

... constructs and think its one of the greatest things about Nu.  Just last night I was working on finding the path that `nix shell` added for a package. I built the command up with:

```nu
$env.path | find nix
# inspect the path, make sure the directory is there, confirm position
# Up-arrow
$env.path | find nix | get 0 | cd $in
```

I just wish I could use it with `let`!

**Comment by NotTheDr01ds** [comment]
I'm a bit surprised at first glance that `| return $in` is even allowed.  I would have thought it would produce the same [`BuiltinCommandInPipeline` error](https://github.com/nushell/nushell/blob/0119534f6193e01f45a0f11f755bfaadddf3b4a0/crates/nu-protocol/src/errors/parse_error.rs#L191) that I get when using it with `let`.

**Comment by fdncred** [comment]
i'd personally like to see all these commands take a pipeline or a parameter. commands like return, let, mut, const, cd, rm, ls, etc.

**Comment by NotTheDr01ds** [comment]
Agreed - Just as long as there are consistent rules for precedence and/or error.

Today, for instance, I believe the most common command that takes a pipeline-or-param is `print`, and in that case having a param will suppress the use of the pipeline (other than `$in`).

```nu
print "param"
# => param
"input" | print
# => input
"input" | print "param"
# => param
```

Perhaps an error would be better, but regardless the rules need to be clear and consistent.

**Comment by sholderbach** [comment]
Related 
- (somewhat different subarea) https://github.com/nushell/nushell/issues/13648
We should based on that treat `return` as a keyword that disallows following pipeline elements.

**Comment by KAAtheWiseGit** [comment]
I agree that having more commands accept pipelined in stuff would be great.

I do think, though, that a silent suppression is a bad idea.  Imagine this:

```Nushell
~> [a b c] | rm d
```

I think it'll be counter-intuitive for this command to only delete `d` and silently ignore all other paths.  Commands such as `rm` and `print` can be additive and delete/print items from both the pipeline and the arguments.  But it can't be used for `return` or `let`.  So, I see throwing an error as the only way to have a consistent behavior across all pipe-in commands and keywords.  But it's possible that commands and keywords should have different semantics when it comes to that.
